backend/controller/scaling/scaling.go

```diff
@@ -2,10 +2,53 @@ package scaling
 
 import (
 	"context"
+	"net/url"
+	"time"
 
-	"github.com/TBD54566975/ftl/internal/model"
+	"github.com/alecthomas/types/optional"
+	"github.com/jpillora/backoff"
+
+	"github.com/TBD54566975/ftl/backend/controller/leases"
+	ftlv1 "github.com/TBD54566975/ftl/backend/protos/xyz/block/ftl/v1"
+	"github.com/TBD54566975/ftl/backend/protos/xyz/block/ftl/v1/ftlv1connect"
+	"github.com/TBD54566975/ftl/internal/log"
+	"github.com/TBD54566975/ftl/internal/rpc"
 )
 
-type RunnerScaling interface {
-	SetReplicas(ctx context.Context, replicas int, idleRunners []model.RunnerKey) error
+type RunnerScaling func(ctx context.Context, endpoint url.URL, leaser leases.Leaser) error
+
+func BeginGrpcScaling(ctx context.Context, url url.URL, leaser leases.Leaser, handler func(ctx context.Context, msg *ftlv1.PullSchemaResponse) error) {
+	leaseTimeout := time.Second * 20
+	for {
+		// Grab a lease to take control of runner scaling
+		lease, leaseContext, err := leaser.AcquireLease(ctx, leases.SystemKey("ftl-scaling", "runner-creation"), leaseTimeout, optional.None[any]())
+		if err == nil {
```

Comment:
In the errors case this should probably only retry if err == ErrConflict, and return an error otherwise, otherwise real errors will get swallowed permanently.